src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java

```diff
@@ -328,11 +327,28 @@ private void processResults(final DataReader dataReader, final BatchInfo batch,
         final String stateMessage = (batch.getState() == BatchStateEnum.Completed) ? null : batch.getStateMessage();
         final String errorMessage = stateMessage == null ? null : Messages.getMessage(getClass(), "batchError",
                 stateMessage);
-
-        final List<Row> rows = dataReader.readRowList(clientBatchInfo.numRows);
+        
+        final int dataReaderRowNumBase = dataReader.getCurrentRowNumber();
+        int totalRowsToReadFromDataReader = clientBatchInfo.numRows;
+        int additionalRowsToReadFromDataReader = 0;
+        int rowCount = 0;
+
+        //Add rows that encountered processing error to the number of rows to be read
+        // from the DataReader
+        while (rowCount < clientBatchInfo.numRows) {
+            boolean conversionSuccessOfRow = isRowConversionSuccessful(dataReaderRowNumBase + rowCount + additionalRowsToReadFromDataReader);
+            if (conversionSuccessOfRow) {
+                rowCount++;
+            } else {
+                additionalRowsToReadFromDataReader++;
+            }
+        }
+        totalRowsToReadFromDataReader += additionalRowsToReadFromDataReader;
```

Comment:
Might be easier to follow if it was initialized here
```suggestion
        final int totalRowsToReadFromDataReader = clientBatchInfo.numRows + additionalRowsToReadFromDataReader;
```

Reply:
good point. I will make that change.